swap/src/protocol/alice/recovery/punish.rs

```diff
@@ -0,0 +1,98 @@
+use crate::bitcoin::{self, ExpiredTimelocks, Txid};
+use crate::database::{Database, Swap};
+use crate::protocol::alice::AliceState;
+use anyhow::{bail, Result};
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    // Errors indicating the the swap can *currently* not be refunded but might be later
+    #[error("Cannot punish because swap is not in a cancelled state. Make sure to cancel the swap before trying to punish or use --force.")]
+    SwapNotCancelled,
+    #[error("The punish timelock has not expired yet because the timelock has not expired. Please try again later")]
```

Comment:
I think you wanted to say something different here: 
```suggestion
    #[error("Cannot punish because the punish timelock has not yet expired. Please try again later")]
```